pygfx/helpers/_lights.py

```diff
@@ -227,4 +227,7 @@ def _update(self, transform: AffineBase):
 
         cone_length = light.distance or 1000
         cone_width = cone_length * math.tan(light.angle)
-        self.scale = (cone_width, cone_width, cone_length)
+
+        self.world.remove_callback(self.update_id)
```

Comment:
```suggestion
        # Temporarily remove callback to avoid infinite recursion when setting `self.local.scale`.
        self.world.remove_callback(self.update_id)
```